server/container/guice/cassandra-guice/src/test/java/org/apache/james/AggregateGuiceModuleTestRule.java

```diff
@@ -0,0 +1,71 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Stream;
+
+import org.junit.rules.RuleChain;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.inject.Module;
+import com.google.inject.util.Modules;
+
+
+public class AggregateGuiceModuleTestRule implements GuiceModuleTestRule {
+
+    public static AggregateGuiceModuleTestRule of(GuiceModuleTestRule... subrule) {
+        return new AggregateGuiceModuleTestRule(Stream.of(subrule));
+    }
+
+    private final Collection<GuiceModuleTestRule> subrule;
+    private final RuleChain chain;
+
+    private AggregateGuiceModuleTestRule(Stream<GuiceModuleTestRule> subrule) {
+        this.subrule = subrule.collect(Guavate.toImmutableList());
+        this.chain = this.subrule.stream().reduce(RuleChain.emptyRuleChain(), RuleChain::around, RuleChain::around);
```

Comment:
you are collecting, then .stream(). I think you can use subrule variable instead of this.subrule.